From 47c9fcc4400d19adcf7e7929d3d105764403d30b Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Thu, 1 May 2014 10:59:50 +0100
Subject: [PATCH 6/6] phonon: Fix PulseAudio card profile list.

If two profiles had the same priority, the map insert would overwrite any
previous profile with that priority.

Simple solutions is to use insertMulti() instead.

Reported by Andrei-Florin Amuraritei
via https://plus.google.com/u/0/+RobinJacobs/posts/GGLemECK6tp
---
 phonon/kcm/audiosetup.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/phonon/kcm/audiosetup.cpp b/phonon/kcm/audiosetup.cpp
index 81a0f53..835799c 100644
--- a/phonon/kcm/audiosetup.cpp
+++ b/phonon/kcm/audiosetup.cpp
@@ -366,7 +366,7 @@ void AudioSetup::updateCard(const pa_card_info *pInfo)
         const quint32 priority = profile->priority;
         const QPair<QString, QString> name(profile->name ? QString::fromUtf8(profile->name) : QString(),
                                            profile->description ? QString::fromUtf8(profile->description) : QString());
-        info.profiles.insert(priority, name);
+        info.profiles.insertMulti(priority, name);
     }
 
     if (pInfo->active_profile)
-- 
1.9.0

